2 Niagara Fundamentals Exercise - Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplexer.java

```diff
@@ -0,0 +1,284 @@
+package com.tridiumuniversity.multiplexer;
+
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.sys.*;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "switchTotal",
+        type = "BInteger",
+        defaultValue = "BInteger.DEFAULT",
+        flags = Flags.READONLY
+)
+@NiagaraProperty(
+        name = "usableSwitches",
+        type = "BInteger",
+        defaultValue = "BInteger.DEFAULT",
+        flags = Flags.READONLY
+)
+@NiagaraProperty(
+        name = "out",
+        type = "BBoolean",
+        defaultValue = "BBoolean.DEFAULT",
+        flags = Flags.SUMMARY | Flags.READONLY
+)
+public class BMultiplexer extends BComponent
+{
+    @Override
+    public void added(Property p, Context cx)
```

Comment:
It wouldn't hurt to wrap the contents of these callback methods in something that checks which Property changed. If another developer came in later and added an `enabled` Property or something like that, we probably wouldn't want to run all of this whenever the `enabled` Property changed.